hail/src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -254,6 +254,30 @@ object Simplify {
 
     case InsertFields(struct, Seq(), _) => struct
 
+    case Let(x, InsertFields(insertedChild, insFields, ord1), InsertFields(Ref(x2, _), fields, ord2)) if x2 == x && {
```

Comment:
I really want tests that would be bad to rewrite and check that we get the expected value. Is it hard to add that?

Reply:
will give it a go, yeah